### DISCUSSION TITLE
一直很好奇有关frpc_windows_amd64.exe的终端图形化是怎么实现的

### DISCUSSION THREAD

**Discussion by cwuom** [natfrp/wiki#385] [通常话题]
很抱歉在这种地方发不是有关产品的问题。
我只是非常好奇这种在终端可以直接运行和点击输出元素的类图形化界面是如何实现的？是使用c++写的吗，我尝试搜索相关内容无果，如果有知道的可以给我一个相关的文档链接吗，我想在自己的项目上套用这种效果，谢谢！
我目前已经有的思路是判断点击做坐标和输出内容的坐标是否相同，但是目前没有找到有效的结果，这个程序是如何做到在缩放和移动窗口位置依旧保证控件可以被正常点击的？还是说只在固定的坐标位置输出内容

[a comment by an automated account is not shown]

**Reply by cwuom** [reply]
是的，谢谢

**Comment by fakeboboliu** [comment]
对于 c++ 使用，请考虑 ncurses，这也是目前最广泛使用的 TUI 库
